tests/integration/targets/web_application/tasks/main.yml

```diff
@@ -0,0 +1,76 @@
+---
+- name: block to ensure there is a cleanup after the tests
+  block:
+  - name: ensure IIS features are installed
+    ansible.windows.win_feature:
+      name: Web-Server
+      state: present
+      include_management_tools: True
+    register: feature_install
+
+  - name: reboot after feature install
+    ansible.windows.win_reboot:
+    when: feature_install.reboot_required
+
+    # may be possible that copy corrupts the file
+  - name: Get iis configuration checksum
+    ansible.windows.win_stat:
+      path: C:\Windows\System32\inetsrv\config\applicationHost.config
+      checksum_algorithm: sha1
+    register: stat_result
+
+  - name: take a copy of the original iis configuration
+    ansible.windows.win_copy:
+      src: C:\Windows\System32\inetsrv\config\applicationHost.config
+      dest: '{{ remote_tmp_dir }}\applicationHost.config'
+      remote_src: yes
+    register: copy_result
+
+  - assert:
+      that:
+        - "stat_result.stat.checksum == copy_result.checksum"
+
+  # Tests
+  - name: run tests on hosts that support it
+    include_tasks: tests.yml
+
+  always:
+  # Cleanup
+  - name: remove test application
+    web_application:
+      state: absent
+      site: "{{ test_site_name }}"
+      name: "{{ test_app_name }}"
+
+  - name: remove test application pool
+    web_app_pool:
+      name: "{{ test_apppool }}"
+      state: absent
+
+  - name: remove test site
+    website:
+      name: "{{ test_site_name }}"
+      state: absent
+
+  - name: restore iis configuration
+    ansible.windows.win_copy:
+      src: '{{ remote_tmp_dir }}\applicationHost.config'
+      dest: C:\Windows\System32\inetsrv\config\applicationHost.config
+      remote_src: yes
+    register: copy_result
+
+  - assert:
+      that:
+        - "stat_result.stat.checksum == copy_result.checksum"
+
+  - name: remove IIS feature if it was installed
+    ansible.windows.win_feature:
+      name: Web-Server
+      state: absent
+      include_management_tools: True
+    when: feature_install is changed
+    register: feature_uninstall
+
+  - name: reboot after removing IIS features
+    ansible.windows.win_reboot:
+    when: feature_uninstall.reboot_required | default(False)
```

Comment:
While normally yes we should clean up everything, to make our tests more efficient we will want to keep this feature. This ensure that each test target doesn't waste time installing the features as a preceeding test will have done so already.

```suggestion
```